src/plugin/file/zip/ui/zipimport.js

```diff
@@ -0,0 +1,347 @@
+};
+goog.inherits(plugin.file.zip.ui.ZIPImportCtrl, os.ui.file.FileImportCtrl);
+
+
+/**
+ * @export
+ */
+plugin.file.zip.ui.ZIPImportCtrl.prototype.finish = function() {
+  var entries = this['files'];
+
+  if (!entries) {
+    var msg = 'No files selected to import from ZIP';
+    os.alert.AlertManager.getInstance().sendAlert(msg, os.alert.AlertEventSeverity.WARNING);
+    return;
+  }
+
+  this.importers_ = [];
+  var unsupported = {};
+
+  for (var i = 0; i < entries.length; i++) {
+    var entry = entries[i];
+    if (entry.enabled === true) {
+      var file = entry.file;
+      var type = file.getType();
+      var ui = this.im_.getImportUI(type);
+      if (ui) {
+        var importer = /** @type plugin.file.zip.ImporterPair */ ({file, ui});
+        this.importers_.push(importer);
+      } else unsupported[type] = true; // store the filetype to report to user later
+    }
+  }
+
+  var keys = Object.keys(unsupported);
+
+  if (keys && keys.length > 0) {
+    var err = 'Unsupported filetype(s).<br />' + keys.join(', ');
+    os.alert.AlertManager.getInstance().sendAlert(err, os.alert.AlertEventSeverity.ERROR);
+  }
+
+  // either close after the chain completes, or close since there is no chain
+  if (this.importers_ && this.importers_.length > 0) this.chain();
+  else this.close(); // for now, close this and rely on the chain; possible future improvement: use slickgrid and show import status
+};
+
+
+/**
+ * Kick off the import of an unzipped file
+ */
+plugin.file.zip.ui.ZIPImportCtrl.prototype.chain = function() {
+  if (typeof this.importers_ == 'undefined' || this.importers_ == null || this.importers_.length == 0) {
+    if (this.curListener_) this.curListener_(); // kill the listener that was created by the previous chain()
+    this.close();
+    return;
+  }
+
+  var onSuccess = function() {
+    // TODO if there are other paths through importers in the future, then update this code
+    //
+    // file successfully read and importer kicked off... so one of two things happened:
+    // 1. asycnhronous window process kicked off...
+    // 2. assume the importer just did some processing then finished; so chain()
+    this.curListener_ = this.scope_.$root.$on(os.ui.WindowEventType.OPEN, this.onWindowOpen_.bind(this));
+
+    var wait = 1350;
+    this.curTimeout_ = this.timeout__(this.onWindowTimeout_.bind(this), wait);
+  };
+
+  var onFailure = function() {
+    this.chain(); // continue
+  };
+
+  this.curImporter_ = this.importers_.splice(0, 1)[0]; // remove the current importer from the queue
+
+  // TODO - Breaking Change -- Fix every place that uses DuplicateImportProcess to utilize events OR alter DuplicateImportProcess
+  // to handle the deferred object in all cases (without breaking any of the current uses).  Currently, the duplicate
+  // dialog is created by setEvent(), so the deferred is not called when that dialog is cancelled.  So for now, we
+  // have to use ImportProcess instead of DuplicateImportProcess here.
```

Comment:
This is probably worth a follow-on ticket.  Basically the issue is that there's no feedback to the creator/caller of Importers which says, "I'm done now" -- with event types for imported, failed to read, failed to load, cancelled, etc.  To add it would break all the existing Importers; since we'd probably modify the interface or add the event(s) to the super class which would need to be overridden properly in some of the edge cases.